subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/TaskHistoryRepository.java

```diff
@@ -28,9 +30,9 @@
 
     interface History {
         @Nullable
-        HistoricalTaskExecution getPreviousExecution();
+        AfterPreviousExecutionState getPreviousExecution();
 
-        CurrentTaskExecution getCurrentExecution();
+        BeforeExecutionState getCurrentExecution();
```

Comment:
I guess this method should be renamed. Now it only returns the state before the current execution.